notebooks/UserSearchPy3.ipynb

```diff
@@ -1328,6 +1331,41 @@
     "lines_to_next_cell": 1
    },
    "outputs": [],
+   "source": [
+    "def prune_hits_to_ignore(full_list, id_to_find):\n",
+    "    # remove vulnerability repos (*-ghsa-*) and archived repos (archive status \n",
+    "    # requires refresh of repository object\n",
+    "    hit_list_1 = [r for r in full_list if (not \"-ghsa-\" in r.repository.name)\n",
+    "                                     and (not r.repository.refresh().archived)\n",
+    "                 ]\n",
+    "    # now eliminate any hits where the search term was not found \"as a word\"\n",
+    "    id_re = re.compile(fr\"\\b{id_to_find}\\b\", re.IGNORECASE)\n",
+    "    hit_list_2 = []\n",
+    "#     print(f\"Checking {len(hit_list_1)} hits\")\n",
+    "    for index, hit in enumerate(hit_list_1):\n",
+    "#         print(f\"  Hit {index} has {len(hit.text_matches)} contexts\")\n",
+    "        for ctxt, context in enumerate(hit.text_matches):\n",
+    "            if id_re.search(context[\"fragment\"]):\n",
+    "                hit_list_2.append(hit)\n",
+    "#                 print(f\"Adding hit {index}; context {ctxt} ({len(hit_list_2)=}): {context['fragment']}\")\n",
+    "                break\n",
+    "            else:\n",
+    "#                 print(f\"ignoring context {context['fragment']}\")\n",
+    "                ...\n",
```

Comment:
This is the new code, which scans the fragments of each "hit" and only keeps those where the search term is a word.